internal/storage/serde.go

```diff
@@ -769,13 +914,28 @@ func newSingleFieldRecordWriter(field *schemapb.FieldSchema, writer io.Writer, o
 		return 1
 	}
 	dim, _ := typeutil.GetDim(field)
+
+	var fieldMetadata arrow.Metadata
+	var arrowType arrow.DataType
+	elementType := schemapb.DataType_None
+
+	if field.DataType == schemapb.DataType_ArrayOfVector {
+		elementType = field.GetElementType()
+		fieldMetadata = arrow.NewMetadata(
+			[]string{"elementType", "dim"},
+			[]string{fmt.Sprintf("%d", int32(elementType)), fmt.Sprintf("%d", dim)},
+		)
+	}
+	arrowType = serdeMap[field.DataType].arrowType(int(dim), elementType)
+
 	w := &singleFieldRecordWriter{
 		fieldId: field.FieldID,
 		schema: arrow.NewSchema([]arrow.Field{
 			{
 				Name:     strconv.Itoa(int(field.FieldID)),
-				Type:     serdeMap[field.DataType].arrowType(int(dim)),
+				Type:     arrowType,
 				Nullable: true, // No nullable check here.
```

Comment:
should we set nullable to false for vector array?

Reply:
Maybe it's just has not be handled as all vectors are not null.